L18n support - Remove Text Select approach

kumarpalsinh25 opened the pull request:
This PR replace `Text-Select` approach with Individual separate strings for simplicity in translation.

gnunicorn commented on app/lib/features/chat/pages/room_profile_page.dart:
```diff
@@ -327,18 +326,18 @@ class _RoomProfilePageState extends ConsumerState<RoomProfilePage> {
                     const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
                 child: Text(
-                  '${L10n.of(context).members} (${list.length})',
+                  L10n.of(context).membersCountRight(list.length),
```
Same as above. let's stick with one...
```suggestion
                  L10n.of(context).membersCount(list.length),
```

gnunicorn commented on app/lib/features/chat/pages/room_profile_page.dart:
```diff
@@ -327,18 +326,18 @@ class _RoomProfilePageState extends ConsumerState<RoomProfilePage> {
             loading: () => Skeletonizer(
               child: Text(
-                '${L10n.of(context).members} (0)',
+                L10n.of(context).membersCountRight(0),
```
```suggestion
                L10n.of(context).membersCount(0),
```

gnunicorn commented on app/lib/features/home/widgets/my_spaces_section.dart:
```diff
@@ -101,22 +98,22 @@ class _NoSpacesWidget extends ConsumerWidget {
               ),
               TextSpan(
-                text: '\t${L10n.of(context).createOrJoinSpace('restText')}',
+                text: '\t${L10n.of(context).spaceShortDescription}',
```
what formatting weirdness is this? why is there a forced tab? Is that how we do design now? Let's remove this, while we are at it.
```suggestion
                text: L10n.of(context).spaceShortDescription,
```

gnunicorn commented on app/lib/features/member/dialogs/show_change_power_level_dialog.dart:
```diff
@@ -114,13 +111,11 @@ class __ChangePowerLevelDialogState extends State<_ChangePowerLevelDialog> {
-                      if (value == null || int.tryParse(value) == null) {
-                        return L10n.of(context)
-                            .youNeedToEnterCustomValueAsNumber;
+                      if ((value == null || int.tryParse(value) == null)) {
+                        return L10n.of(context).youNeedToEnterCustomValueAsNumber;
```
I think a recent update reformatted this code and now this reverts that ...
```suggestion
                      if (value == null || int.tryParse(value) == null) {
                        return L10n.of(context).youNeedToEnterCustomValueAsNumber;
```

gnunicorn commented on app/lib/features/member/dialogs/show_change_power_level_dialog.dart:
```diff
@@ -140,30 +135,32 @@ class __ChangePowerLevelDialogState extends State<_ChangePowerLevelDialog> {
+              if (freshMemberStatus == currentMemberStatus) {
+                // nothing to do, all the same.
+                Navigator.pop(context, null);
+                return;
+              }
+              int? newValue;
+              if (currentMemberStatus == 'Admin') {
+                newValue = 100;
+              } else if (currentMemberStatus == 'Mod') {
+                newValue = 50;
+              } else if (currentMemberStatus == 'Regular') {
+                newValue = 0;
+              } else {
+                newValue = customValue ?? 0;
+              }
 
-            if (currentPowerLevel == newValue) {
-              // nothing to be done.
-              newValue = null;
-            }
+              if (currentPowerLevel == newValue) {
+                // nothing to be done.
+                newValue = null;
+              }
 
-            Navigator.pop(context, newValue);
+              Navigator.pop(context, newValue);
+              return;
+            }
```
same here ... there was a change made to remove that extra indentation level this PR re-introduces again now ... Probably shouldn't be happening.

gnunicorn commented on app/lib/features/onboarding/pages/intro_page.dart:
```diff
@@ -49,7 +49,7 @@ class IntroPage extends StatelessWidget {
             children: <TextSpan>[
               TextSpan(
-                text: '\n${L10n.of(context).appName('')}!',
+                text: '\n${L10n.of(context).acter}!',
```
```suggestion
                text: L10n.of(context).acter,
```

same as before...

gnunicorn commented on app/lib/features/onboarding/pages/intro_page.dart:
```diff
@@ -82,11 +82,11 @@ class IntroPage extends StatelessWidget {
             RichText(
               text: TextSpan(
-                text: L10n.of(context).introPageDescription('desc1'),
+                text: L10n.of(context).introPageDescription1,
```
this desc1, desc2, desc3 doesn't really work well for me ... I am wondering if we instead want to offer some  more semantic info or allow for a more comprehensive form of allowing to write entire paragraphs ... but it's fine for now

gnunicorn commented on app/lib/features/pins/widgets/pin_list_item.dart:
```diff
@@ -33,7 +33,7 @@ class PinListItemById extends ConsumerWidget {
       ),
       error: (err, st) => Text(
-        '${L10n.of(context).error} ${L10n.of(context).loading('').toLowerCase()}: ${err.toString()}',
+        '${L10n.of(context).error} ${L10n.of(context).loading.toLowerCase()}: ${err.toString()}',
```
why isn't this the same `failedToLoad(error)` as used elsewhere?

```suggestion
L10n.of(context).failedToLoad(err.toString())
```

gnunicorn commented on app/lib/features/settings/pages/labs_page.dart:
```diff
@@ -29,11 +29,11 @@ class SettingsLabsPage extends ConsumerWidget {
                 SettingsTile.switchTile(
                   title: Text(
-                    '${L10n.of(context).encrypted} ${L10n.of(context).spaces('').toLowerCase()}',
+                    '${L10n.of(context).encrypted} ${L10n.of(context).spaces.toLowerCase()}',
```
grammar: Order of words, forced letter casing ... all that is language specific:

```suggestion
                    L10n.of(context).encryptedSpaces,
```

gnunicorn commented on app/lib/features/settings/super_invites/widgets/redeem_token.dart:
```diff
@@ -54,7 +54,7 @@ class _RedeemTokenConsumerState extends ConsumerState<RedeemToken> {
     final superInvites = ref.read(superInvitesProvider);
     try {
-      EasyLoading.show(status: '${L10n.of(context).redeem('withIng')} $token');
+      EasyLoading.show(status: '${L10n.of(context).redeeming} $token');
```
```suggestion
      EasyLoading.show(status: L10n.of(context).redeeming(token));
```

gnunicorn commented on app/lib/features/space/widgets/links_card.dart:
```diff
@@ -34,7 +34,7 @@ class LinksCard extends ConsumerWidget {
                     return [
                       Text(
-                        L10n.of(context).thereAreNoItemInThisSpace('pins'),
+                        L10n.of(context).thereAreNoPinsInThisSpace,
```
May I suggest to stick with simple keys rather than writing out entire sentences...?!?
```suggestion
                        L10n.of(context).noPinsInSpace,
```

gnunicorn commented on app/lib/features/tasks/dialogs/create_task_list_sheet.dart:
```diff
@@ -42,45 +42,50 @@ class _CreateTaskListSheetConsumerState
-      }
-      final taskListId = await taskListDraft.send();
-      // reset providers
+    if (_formKey.currentState!.validate()) {
+      DefaultDialog(
```
indentation formatting ....

gnunicorn commented on app/lib/features/tasks/widgets/due_chip.dart:
```diff
@@ -81,7 +81,7 @@ class _DueChipState extends State<DueChip> {
         // FIXME: tooltip to show the full date?
         label ??
-            '${L10n.of(context).due('')}: ${DateFormat(DateFormat.YEAR_MONTH_WEEKDAY_DAY).format(dueDate!)}',
+            '${L10n.of(context).due}: ${DateFormat(DateFormat.YEAR_MONTH_WEEKDAY_DAY).format(dueDate!)}',
```
this should be an input as well.
```suggestion
           L10n.of(context).dueDate(DateFormat(DateFormat.YEAR_MONTH_WEEKDAY_DAY).format(dueDate!)),
```

And arguably the dueDate format might also be something we want to localize ... but I suppose that happens in the DateFormat-lib rather than us giving the format in the language-file...?!?

kumarpalsinh25 replied:
Created separate ticket to manage this.
https://github.com/acterglobal/a3/issues/1577

gnunicorn commented on app/lib/features/tasks/widgets/task_info.dart:
```diff
@@ -179,7 +179,12 @@ class _TaskTitleState extends State<TaskTitle> {
                 ),
               ),
-              onFieldSubmitted: _handleSubmit,
+              onFieldSubmitted: (value) async {
+                if (_formKey.currentState!.validate()) {
+                  _formKey.currentState!.save();
+                  await _handleSubmit();
+                }
+              },
```
this was also changed in the formatting-PR and should stay that way ...

gnunicorn commented on app/lib/features/tasks/widgets/task_list_card.dart:
```diff
@@ -216,7 +214,12 @@ class _InlineTaskAddState extends State<_InlineTaskAdd> {
           ),
         ),
-        onFieldSubmitted: _handleSubmit,
+        onFieldSubmitted: (value) {
```
as above...

gnunicorn reviewed (commented):
Thanks heaps.

There's a few clashes that we probably want to address. At least one is due to an incorrect merge of what is currently in `main` mixing up some indentation, some other are not yet complete conversion to the language format ....

kumarpalsinh25 commented:
> At least one is due to an incorrect merge of what is currently in `main` mixing up some indentation, 

Fixed incorrect merged related changes.

> some other are not yet complete conversion to the language format ....

Resolve point which are mentioned in this PR review comments. I still feel that there is scope of improvements in string key-naming and other relevant changes. I have added this point in https://github.com/acterglobal/a3/issues/1567 for clean this with separate PR. :)

gnunicorn commented on app/lib/features/home/widgets/my_spaces_section.dart:
```diff
@@ -30,7 +30,7 @@ class MySpacesSection extends ConsumerWidget {
           ),
         ),
-        spaces.isEmpty
+        !spaces.isEmpty
```
This seems wrong

kumarpalsinh25 replied:
It was change when I was testing empty data state locally and was committed by mistake by already fixed on last committ.
https://github.com/acterglobal/a3/pull/1569/commits/765118d855638fb96f461a3844d0e9f12bacae67
